### PR TITLE
Refer Antlr file using specific branch

### PR DESCRIPTION
**Pull request by ykmr1224**
### Description
- Refer Antlr file using specific tag/branch (Spark -> v3.5.1, Flint -> 0.5)
- Disable auto download  

### Related Issues
https://github.com/opensearch-project/sql/pull/2890#issuecomment-2264199142
https://github.com/opensearch-project/sql/issues/3022

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
 - [ ] New functionality has javadoc added.
 - [ ] New functionality has a user manual doc added.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [x] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/sql/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: async-query-core/src/main/antlr/SqlBaseLexer.g4 [hunk 1]
```diff
@@ -18,7 +18,7 @@ lexer grammar SqlBaseLexer;
 
 @members {
   /**
-   * When true, parser should throw ParseException for unclosed bracketed comment.
+   * When true, parser should throw ParseExcetion for unclosed bracketed comment.
```
Comment (YANG-DB) [inline async-query-core/src/main/antlr/SqlBaseLexer.g4:21]: Should it be Exception ?

**Review by dai-chen — DISMISSED** [review]
Thanks for the changes!

I'm thinking how we can ensure that we remember to update this version whenever a new release branch is created in `opensearch-spark` repo. cc: @penghuo

**Comment by ykmr1224** [comment]
> I'm thinking how we can ensure that we remember to update this version whenever a new release branch is created in `opensearch-spark` repo. cc: @penghuo

Should we have a check list for creating new release branch in `opensearch-spark` repo?

**Review by dai-chen — COMMENTED** [review]
I'm thinking whether tracking one Flint Spark branch will be a problem or not.
I think the ideal case SQL `main` tracks Spark `main` and SQL `2.x` tracks Spark `0.5`? However, we cannot do this because we always backport PR?

**Comment by ykmr1224** [comment]
> I'm thinking whether tracking one Flint Spark branch will be a problem or not. I think the ideal case SQL `main` tracks Spark `main` and SQL `2.x` tracks Spark `0.5`? However, we cannot do this because we always backport PR?

I think the problem is that we tend to automatically include the grammar change in the PRs even if the change is not related to the grammar. We'd want to separate Grammar PRs and don't backport them.

Options came up to my mind are:
1. Do not automatically update Grammar files.
2. Separate the grammar files for repository and ones for build, and only update the ones for build during build. And add build target to update grammar file for repository.
3. Manually avoid including the Grammar files to PRs.

**Review by manasvinibs — APPROVED** [review]

**Review by seankao-az — APPROVED** [review]

**Pull request merged**